quixstreams/processing_context.py

```diff
@@ -0,0 +1,80 @@
+import dataclasses
+import logging
+import time
+from typing import Optional
+
+from quixstreams.checkpointing import Checkpoint
+from quixstreams.exceptions import QuixException
+from quixstreams.rowconsumer import RowConsumer
+from quixstreams.rowproducer import RowProducer
+from quixstreams.state import StateStoreManager
+
+__all__ = ("ProcessingContext",)
+
+logger = logging.getLogger(__name__)
+
+
+class CheckpointNotInitialized(QuixException): ...
+
+
+@dataclasses.dataclass
+class ProcessingContext:
```

Comment:
This class is fairly coupled with checkpoints, wonder if it would make sense to have them grouped together more (no functional changes, just organizational)?

Reply:
What exactly do you mean by grouping `ProcessingContext` and `Checkpoint` classes?

Reply:
For posterity: `ProcessingContext` uses `Checkpoint` as a dependency, but they don't need to be in the same modules.